OpenRA.Mods.Common/AI/States/AirStates.cs

```diff
@@ -63,13 +63,14 @@ protected static Actor FindDefenselessTarget(Squad owner)
 		protected static CPos? FindSafePlace(Squad owner, out Actor detectedEnemyTarget, bool needTarget)
 		{
 			var map = owner.World.Map;
+			var dangerRadius = owner.Bot.Info.DangerScanRadius;
 			detectedEnemyTarget = null;
-			var x = (map.MapSize.X % DangerRadius) == 0 ? map.MapSize.X : map.MapSize.X + DangerRadius;
-			var y = (map.MapSize.Y % DangerRadius) == 0 ? map.MapSize.Y : map.MapSize.Y + DangerRadius;
+			var x = (map.MapSize.X % dangerRadius) == 0 ? map.MapSize.X : map.MapSize.X + dangerRadius;
+			var y = (map.MapSize.Y % dangerRadius) == 0 ? map.MapSize.Y : map.MapSize.Y + dangerRadius;
 
-			for (var i = 0; i < x; i += DangerRadius * 2)
+			for (var i = 0; i < x; i += dangerRadius * 2)
```

Comment:
Manually chunking the map like this is really awful :sad:

In the future we could win a lot of simplicity and perf by maintaining spatially partitioned maps of targetability and danger for each squad class.